### DISCUSSION TITLE
Change inherited points to automation

### DISCUSSION THREAD

**Discussion by ghost** [ppy/osu#13141] [Ideas]
Original post: https://osu.ppy.sh/community/forums/topics/1119393

**Automation** is a tool in DAWs to change parameters over time. (https://www.izotope.com/en/learn/what-is-mix-automation.html)

Automation would be much more practical and fast.

In osu!lazer's editor, there is a waveform where the automation could be shown. 

It would be much easier to see than the timing setup panel since there is a different automation for each parameter.

The only flaw with this design is if you have really big changes.

(Parameters: Slider Velocity, Hitsounds Volume, _Maybe AR and CS_)

The other parameters would be represented in a different way. _(I will say how later)_

**Comment by smoogipoo** [comment]
I'm not an audio guy so pardon my noobness. But I'm a little bit confused by this issue, because lazer has already removed the concept of inherited timing points.

Are you just asking for better display/controlling the automation via the timeline, or are you asking for _more_ automation (fade, etc)?

**Comment by peppy** [comment]
Basically asking for transforms/tweens (what "automation" means in music production). Which is something we have planned for sure.

**Comment by ghost** [comment]
I didn't know that lazer had removed them but I just wanted to give this idea. I think it could be a great substitute. 

For the flaw I gave: _"The only flaw with this design is if you have really big changes."_ You just need a zoom option.

I don't know what a tween is, but from what I saw in Wikipedia that's similar to what I mean.

> Are you just asking for better display/controlling the automation via the timeline, or are you asking for more automation (fade, etc)?

I'm sorry I don't know what you mean. 

And for the other parameters (Samples and Kiai Mode) using automation wouldn't make sense.

For Samples, I will make another issue (Hitsound system).

For Kiai mode, I don't know...

**The rest of this comment it's a suggestion for how it would work.**

To show the automation you could have it in the compose section on top of the waveform.

To enable or disable it you could have a toggle on the left panel. 

When it's enabled it could show a drop-down menu to choose which parameter to change.

To create a point left-click. When you hover over it it shows which value it represents. Right-click to change its type _(I'll comment more about this later)_. To edit its value drag it up or down/double click it to type the value. To change where it is drag it left or right. Delete key to delete it.

This is very similar to how it works in DAWs.

**Comment by ghost** [comment]
![standard](https://user-images.githubusercontent.com/59707548/89544563-30475b00-d7fa-11ea-8efd-f6486671d3bb.png)
These automations were made with the **standard/linear point**.
![Step](https://user-images.githubusercontent.com/59707548/89544623-45bc8500-d7fa-11ea-8cba-afb026e223a7.png)
This one with the **step point**.

These points are the only ones that I think are useful for osu!. 

Examples: Standard/Linear - Volume | Step - Slider Velocity

You can also manipulate how the values change between points with the linear type. (You drag the point in the middle up or down | _1st Automation_)
